client/my-sites/checkout/src/components/checkout-sidebar-plan-upsell/index.tsx

```diff
@@ -114,10 +209,27 @@ export function CheckoutSidebarPlanUpsell() {
 		if ( isFormLoading ) {
 			return;
 		}
-		const newPlan = {
-			product_slug: biennialVariant.productSlug,
-			product_id: biennialVariant.productId,
-		};
+
+		let newPlan;
```

Comment:
Not a blocker but using `let` is risky for a couple reasons:

1. It allows future developers to continue to redefine this variable anywhere below this line, making it hard to tell what its value will be when we see an instance of it in the code; we have to read all the code between that use-case and this definition.
2. It means several different definitions might take place, meaning that the order of those definitions is critical to understand when reading the code. You have to always consider what might come below the current line any time there's a redefinition of the variable.

Ternaries are a good pattern to use to allow using `const` for a variable that can have two possible values, but here you have three possible values and ternaries are also not great for long values or complex conditions. In this case, you could use an IIFE instead:

```js
const newPlan = (() => {
		if ( currentVariant.termIntervalInMonths === 1 ) {
			return {
				product_slug: annualVariant.productSlug,
				product_id: annualVariant.productId,
			};
		}

		if ( currentVariant.termIntervalInMonths === 12 ) {
			return {
				product_slug: biennialVariant.productSlug,
				product_id: biennialVariant.productId,
			};
		}
})()

if ( ! newPlan ) {
	return;
}
```

Reply:
Updated to use the IIFE, thanks for the example!